test/test_transforms_v2_refactored.py

```diff
@@ -0,0 +1,326 @@
+import inspect
+import re
+from typing import get_type_hints
+from unittest import mock
+
+import PIL.Image
+import pytest
+
+import torch
+from common_utils import cache, cpu_and_gpu, make_bounding_box, make_image
+from torch.testing import assert_close
+from torch.utils._pytree import tree_map
+from torchvision import datapoints
+from torchvision.transforms.v2 import functional as F
+from torchvision.transforms.v2.utils import is_simple_tensor
+
+
+def _to_tolerances(maybe_tolerance_dict):
+    default_tolerances = dict(rtol=None, atol=None)
+    if not isinstance(maybe_tolerance_dict, dict):
+        return default_tolerances
+
+    missing = default_tolerances.keys() - maybe_tolerance_dict.keys()
+    if missing:
+        raise pytest.UsageError("ADDME")
+
+    extra = maybe_tolerance_dict.keys() - default_tolerances.keys()
+    if extra:
+        raise pytest.UsageError("ADDME")
+
+    return maybe_tolerance_dict
+
+
+def _check_kernel_cuda_vs_cpu(kernel, tolerances, input, *args, **kwargs):
+    if input.device.type != "cuda":
+        return
+
+    input_cuda = input.as_subclass(torch.Tensor)
+    input_cpu = input_cuda.to("cpu")
+
+    actual = kernel(input_cuda, *args, **kwargs)
+    expected = kernel(input_cpu, *args, **kwargs)
+
+    assert_close(actual, expected, **tolerances, check_device=False)
+
+
+@cache
+def _script(fn):
+    try:
+        return torch.jit.script(fn)
+    except Exception as error:
+        raise AssertionError(f"Trying to `torch.jit.script` '{fn.__name__}' raised the error above.") from error
+
+
+def _check_kernel_scripted_vs_eager(kernel, tolerances, input, *args, **kwargs):
+    kernel_scripted = _script(kernel)
+
+    input = input.as_subclass(torch.Tensor)
+    actual = kernel_scripted(input, *args, **kwargs)
+    expected = kernel(input, *args, **kwargs)
+
+    assert_close(actual, expected, **tolerances)
+
+
+def _unbatch(batch, *, data_dims):
+    if isinstance(batch, torch.Tensor):
+        batched_tensor = batch
+        metadata = ()
+    else:
+        batched_tensor, *metadata = batch
+
+    if batched_tensor.ndim == data_dims:
+        return batch
+
+    return [
+        _unbatch(unbatched, data_dims=data_dims)
+        for unbatched in (
+            batched_tensor.unbind(0) if not metadata else [(t, *metadata) for t in batched_tensor.unbind(0)]
+        )
+    ]
```

Comment:
> One of design principles of transforms v2 supports arbitrary batch sizes

You mean arbitrary number of leading dimension?

Do we ever call this with more than 1 batch dim? (hard to figure out right now because this is actually never called)

Reply:
Right now, no. The parametrization is insufficient as explained in https://github.com/pytorch/vision/pull/7562#discussion_r1204157444. Let me send a patch to include this so we have a better base for discussion.